spring-cloud-kubernetes-examples/kubernetes-leader-election-example/README.md

```diff
@@ -0,0 +1,83 @@
+# Spring Cloud Kubernetes leader election example
+
+## Setting up the Environment
+
+This example uses a Fabric8 Maven Plugin to deploy an application to a Kubernetes cluster.
+To try it locally, download and install `[Minikube](https://kubernetes.io/docs/getting-started-guides/minikube/)`.
+
+Once Minikube is downloaded, start it with the following command:
+```
+minikube start
+```
+
+And configure environment variables to point to the Minikube's Docker daemon:
+```
+eval $(minikube docker-env)
+```
+
+## Overview
+
+Spring Cloud Kubernetes leader election mechanism implements leader election API of Spring Integration using Kubernetes ConfigMap.
+Multiple instances of the same application compete for a leadership of a specified role.
+But only one of them can become a leader and receive an `OnGrantedEvent` application event with a leadership `Context`.
+All the instances periodically try to get a leadership and whichever comes first - becomes a leader.
+The new leader will remain until either its instance disappears from the cluster or it yields its leadership.
+Once the leader is gone, any of the existing instances can become a new leader (even the previous leader if it yielded leadership but stayed in the cluster). 
+And finally, if the leadership is yielded or revoked for some reason, the old leader receives `OnRevokedEvent` application event.
+
+## Example application usage
+
+To begin with, build and deploy the application:
+```
+mvn clean package fabric8:deploy -Pkubernetes
+```
+
+This will deploy a single application instance to the cluster and that instance will automatically become a leader.
+
+Create an environment variable for an easier application access:
+```
+SERVICE_URL=$(minikube service kubernetes-leader-election-example --url)
+```
+
+Get leadership information:
+```
+curl $SERVICE_URL
+```
+
+You should receive a message like this:
+```
+I am 'kubernetes-leader-election-example-1234567890-abcde' and I am the leader of the 'world'
+```
+
+Yield the leadership:
```

Comment:
I'm confused about what happens here as when you query `curl $SERVICE_URL`, you then get the message `I am 'kubernetes-leader-election-example-1234567890-abcde' and I am the leader of the 'world'` which means that I'm already a leader. So, will happen then when you issue the `curl -x PUT` command ?

Reply:
If it produces the leadership, then yur first `curl GET` request should not return `I am 'kubernetes-leader-election-example-1234567890-abcde' and I am the leader of the 'world'` but instead `I am 'kubernetes-leader-election-example-1234567890-abcde' and I am NOT leader of the 'world'`